app/main.py

```diff
@@ -1,3 +1,38 @@
-def shop_trip():
-    # write your code here
-    pass
+from app.data_config import customers, shops, FUEL_PRICE
+
+
+def shop_trip() -> None:
+    for customer in customers.values():
+        print(f"{customer.name} has {customer.money} dollars")
+        home_location = customer.location
+        min_trip_costs = float("inf")
+        cheapest_shop = None
+
+        for shop in shops.values():
+            shop_distance = shop.calculate_distance_to_shop(customer.location)
+            fuel_expense = customer.car.calculate_fuel_expense(shop_distance)
+            products_cost = shop.calculate_products_cost(customer.product_cart)
+            trip_costs = round(
+                fuel_expense * FUEL_PRICE * 2 + products_cost,
+                2)
```

Comment:
In case you do not place any arguments on the same line with opening parenthesis, you closing parenthesis should not have any arguments on the same line as well.